JASP-Engine/JASP/R/abtestbayesian.R

```diff
@@ -16,283 +16,191 @@
-    # Create status object and do error checking; exit the analysis if any errors are found
-    #
-    # Args:
-    #   dataset: dataset input by user
-    #   perform: 'run' or 'init'
-    #   options: a list of user options
-    #
-    # Return:
-    #   A status object containing "ready", "error", "errorMessage"
+    return()
+  }
 
-    status <- list(ready = TRUE, error = FALSE, errorMessage = "")
+  output.rows <- list()
+  rowCount    <- 0
 
-    if (options$n1 == "" || options$n2 == "" || options$y1 == "" || options$y2 == "") {
-        status$ready <- FALSE
+  if (orEqualTo1Prob > 0) {
+    rowCount = rowCount + 1
+    output.rows[[rowCount]] <- list(
+      "Models"    = "Log odds ratio = 0",
+      "BF"        = 1.00,
+      "P(M|data)" = ab_obj$post_prob[["H0"]],
+      "P(M)"      = orEqualTo1Prob
+    )
+  }
+
+  if (orGreaterThan1Prob > 0) {
+    rowCount = rowCount + 1
+    output.rows[[rowCount]] <- list(
+      "Models"    = "Log odds ratio > 0",
+      "BF"        = ab_obj$bf[["bfplus0"]],
+      "P(M|data)" = ab_obj$post_prob[["H+"]],
+      "P(M)"      = orGreaterThan1Prob
+    )
+  }
+
+  if (orLessThan1Prob > 0) {
+    rowCount = rowCount + 1
+    output.rows[[rowCount]] <- list(
+      "Models"    = "Log odds ratio < 0",
+      "BF"        = ab_obj$bf[["bfminus0"]],
+      "P(M|data)" = ab_obj$post_prob[["H-"]],
+      "P(M)"      = orLessThan1Prob
+    )
+  }
+
+  if (orNotEqualTo1Prob > 0) {
+    rowCount = rowCount + 1
+    output.rows[[rowCount]] <- list(
+      "Models"    = "Log odds ratio ≠ 0",
+      "BF"        = ab_obj$bf[["bf10"]],
+      "P(M|data)" = ab_obj$post_prob[["H1"]],
+      "P(M)"      = orNotEqualTo1Prob
+    )
+  }
+
+  if (options$bayesFactorOrder == "bestModelTop") {
+    ordered       <- output.rows[order(sapply(output.rows, "[[", "P(M|data)"), decreasing = TRUE)]
+    best_model_bf <- ordered[[1]]$BF
+    output.rows   <- list()
+
+    for (r in 1:rowCount) {
+      ordered[[r]]$BF  <- ordered[[r]]$BF / best_model_bf
     }
+    output.rows   <- ordered
+  }
 
-    # customChecks <- list()
-    #
-    # .hasErrors(dataset = dataset, perform = perform,
-    # 	type=c("infinity", "observations", "variance"), custom = customChecks,
-    # 	infinity.target = c(options$covariates, options$dependent, options$wlsWeight),
-    # 	observations.target = options$dependent, observations.amount = paste("<", length(options$modelTerms) + 1),
-    # 	variance.target = c(options$covariates, options$dependent),
-    # 	exitAnalysisIfErrors = TRUE)
-
-    return(status)
+  for (r in 1:rowCount) {
+    if        (options$bayesFactorType == "BF01") {
+      output.rows[[r]]$BF <- 1 / output.rows[[r]]$BF
+    } else if (options$bayesFactorType == "LogBF10") {
+      output.rows[[r]]$BF <- base::log(output.rows[[r]]$BF)
+    }
+  }
+
+  abTestBayesianTable$addRows(output.rows)
 }
 
 
-.calcModel.abTest <- function(dataset, status, options) {
+.calcModel.abTest <- function(dataset, options) {
```

Comment:
You should store this in a jaspState; right now it's being computed every time and it's making the analysis slower than it needs to be

Reply:
@akashrajkn do you have time to incorporate this? Or shall I do it and merge this now

Reply:
@akashrajkn I'll merge this tomorrow and implement this myself unless you don't want me to